base/set.jl

```diff
@@ -191,6 +191,69 @@ function unique(f::Callable, C)
 end
 
 """
+    unique!(A::AbstractVector)
+
+Remove duplicate items as determined by [`isequal`](@ref), then return the modified `A`.
+
+```jldoctest
+julia> unique!([1, 1, 1])
+1-element Array{Int64,1}:
+ 1
+
+julia> A = [7, 3, 2, 3, 7, 5];
+
+julia> unique!(A);
+
+julia> A
+4-element Array{Int64,1}:
+ 7
+ 3
+ 2
+ 5
+```
+"""
+function unique!(A::AbstractVector)
+    isempty(A) && return A
+
+    try
+        sorted = issorted(A) || issorted(A, rev=true)
+    catch
+        sorted = false
+    end
+
+    if sorted
+        # If A is sorted, then we only need to keep track of one element and add that to A
+        # every time that we see a new element.
+        idxs = eachindex(A)
+        y = first(A)
+        state = start(idxs)
+        for i in idxs
+            x = A[i]
+            if !isequal(x, y)
+                j, state = next(idxs, state)
+                y = A[j+1] = x
```

Comment:
It doesn't sound right to do `+1` here. Wouldn't it be better to add `_, state = next(idxs, state)` right before the loop instead?